cmds/replay.c

```diff
@@ -586,19 +598,39 @@ void get_argspec_string(struct uftrace_task_reader *task,
 
 			if (!memcmp(str, &null_str, sizeof(null_str)))
 				print_args("NULL");
-			else if (needs_escape) {
+			else if (needs_json) {
 				char *p = str;
 				print_args("\\\"");
 				while (*p) {
 					char c = *p++;
-					print_escaped_char(c);
+					print_json_escaped_char(c);
 				}
 				print_args("\\\"");
 			}
 			else {
 				print_args("%s", color_string);
-				print_args("\"%.*s\"", slen + newline, str);
-				print_args("%s", color_reset);
+				print_args("\"");
+
+                                char *p = str;
+                                while (*p) {
+                                        char c = *p++;
+                                        if (c & 0x80) {
+                                                break;
+                                        }
+                                }
+                                if (*p) {
+                                        print_args("%.*s", slen + newline, str);
```

Comment:
Could you please add a comment that it assumes UTF-8 string here?

Reply:
Okay. I will add comment

Reply:
We have a loose 80 character limitation in a line.  Could you please make it a multi-line comment?